src/EnergyPlus/HybridEvapCoolingModel.cc

```diff
@@ -0,0 +1,2389 @@
+			count_EnvironmentConditionsMetOnce = 0;
+			count_EnvironmentConditionsNotMet = 0;
+			count_SAHR_OC_MetOnce = 0;
+			count_SAT_OC_MetOnce = 0;
+			count_DidWeMeetLoad = 0;
+			count_DidWeNotMeetLoad = 0;
+			// vector below used store the modes in each timestep that don't meet humidity or temperature limits, used in warnings
+			std::vector<int> temp(25);
+			SAT_OC_MetinMode_v= temp;
+			SAHR_OC_MetinMode_v= temp;
+			
+			int MAXIMUM_OPERATIONAL_SETTINGS = 5;
+			ModeCounter = 0;
+			//!!!!!!!!!!!!!!!!!!!!!!!!! this debug code will be removed nearer the code freeze, please don't comment on it, it will be gone.
+			DebugBreak = 12; // remove
+	
+			CurrentOperatingSettings.resize(5);
+
+			InitializeModelParams();
+		}
+		
+		void Model::ResetOutputs()
+		{
+			UnitTotalCoolingRate = 0;
+			UnitTotalCoolingEnergy = 0;
+			UnitSensibleCoolingRate = 0;
+			UnitSensibleCoolingEnergy = 0;
+			UnitLatentCoolingRate = 0;
+			UnitLatentCoolingEnergy = 0;
+			SystemTotalCoolingRate = 0;
+			SystemTotalCoolingEnergy = 0;
+			SystemSensibleCoolingRate = 0;
+			SystemSensibleCoolingEnergy = 0;
+			SystemLatentCoolingRate = 0;
+			SystemLatentCoolingEnergy = 0;
+			UnitTotalHeatingRate = 0;
+			UnitTotalHeatingEnergy = 0;
+			UnitSensibleHeatingRate = 0;
+			UnitSensibleHeatingEnergy = 0;
+			UnitLatentHeatingRate = 0;
+			UnitLatentHeatingEnergy = 0;
+			SystemTotalHeatingRate = 0;
+			SystemTotalHeatingEnergy = 0;
+			SystemSensibleHeatingRate = 0;
+			SystemSensibleHeatingEnergy = 0;
+			SystemLatentHeatingRate = 0;
+			SystemLatentHeatingEnergy = 0;
+			SupplyFanElectricPower = 0;	   
+			SupplyFanElectricEnergy = 0;    
+			SecondaryFuelConsumptionRate = 0;
+			SecondaryFuelConsumption = 0;   
+			ThirdFuelConsumptionRate = 0;   
+			ThirdFuelConsumption = 0;	   
+			WaterConsumptionRate = 0;	   
+			WaterConsumption = 0;		   
+			ExternalStaticPressure = 0;	 
+			QLatentZoneOut = 0;
+			QSensZoneOut = 0;
+			//debug
+			//RawHR = 0;
+		}
+
+		void Model::InitializeModelParams()
+		{
+			// SUBROUTINE INFORMATION:
+			//       AUTHOR         Spencer Maxwell Dutton
+			//       DATE WRITTEN   October 2017 
+			//       MODIFIED       
+			//       RE-ENGINEERED  na
+
+			// PURPOSE OF THIS SUBROUTINE:
+			// Reset calculation values 
+
+			// METHODOLOGY EMPLOYED:
+			// 
+
+			// REFERENCES:
+			// na
+
+			// Using/Aliasing
+			ResetOutputs();
+			PrimaryMode = 0;
+			PrimaryModeRuntimeFraction = 0;
+			optimal_EnvCondMet = false;
+			Tsa = 0;
+			//reset the power use to a high value, this is replaced during the calculation keeping the "best" setting. 
+			//pOptimal.ElectricalPower = IMPLAUSIBLE_POWER;
+			RunningPeakCapacity_EnvCondMet = false;
+			Settings.clear();
+		}
+
+		Model::~Model()                 
+		{
+			for (auto iter = SolutionSpaces.begin(); iter != SolutionSpaces.end(); ++iter) {
+				CModeSolutionSpace* p = (*iter);
+				delete p;
```

Comment:
I don't like what I'm seeing here.  Calling `delete` is a big red flag here in C++.  My suggestion is to quickly figure out an alternative solution to manual memory management before any additional review.

Reply:
This was old obsolete code left over by mistake, i'm sorry. I have pushed up code where that obsolete code is removed. 

In the remainder of the code here is now only the occasional use of "safe" pointers 

std::shared_ptr< CSetting > pCandidateSetting = std::shared_ptr< CSetting >(new CSetting());

Reply:
Sorry, but I don't think the alternative code is much better. Why is this needed? There still appears to be manual memory management in various forms throughout this code.

Reply:
Unless I messed up the push up there should be no deletes left if the  code only occasional use of safe pointers that are I understand  also used in the rest of the code base. It found it necessary to use safe pointers because the number of settings the model looks at changes radically depending on how you specify the model, it needs the flexibility afforded by the safe pointers.

Reply:
GitHub is already starting to drag, so I think we're already on our way to another angry unicorn. The use of smart pointers without more justification (why shared instead of unique, for example) is a reddish flag for me, and then were non-trivial destructors that I can no longer find. Maybe those were also obsolete, too? In that case, the branch isn't really ready for review.

Reply:
@jasondegraw. The number of potential settings a system can be in can vary dramatically depending on the type of system being modeled. A setting is a unique combination of, the mode the unit is in, the outside air fraction, and supply air mass flow rate. The way this is dealt with in the code is we keep a very large list of all the possible setting this model could be in, that doesn't change once its setup.  Then, when the system is operating one or more of these setting are chosen to be kept in the OperatingSettings list (the setting the unit is actually using this time step). This means we have a single resource that is being referred to in more than one location. So, I feel like the use of a shared smart pointers here is justified, or at least I can not think of a more elegant solution.